### PR TITLE
add im2col app example

### PR DESCRIPTION
**Pull request by TommiTerza**
These modifications allow the test, verification and performance analysis of the im2col function, implemented in C code, with the ultimate goal of testing the capabilities of the multichannel DMA.
The python script "verification_script.py" generates a random tensor with configurable channels and dimensions, together with a "golden result" used for the verification of the C code, i.e. the output matrix of the im2col.

### CODE REVIEW

File: sw/applications/example_im2col/im2col_lib.c [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+                size_transfer = 0;
+                minimum = 0;
+                start_max = 0;
+                start_min = 0;
+                stray_elements = 0;
+                last_position = 0;
+                tmp_pad = 0;
+
+                #if DEBUG==2    
+                printf("\nim_row: %d, im_col: %d", im_row, im_col);
```
[the remark this answers is not in the document]
Comment (TommiTerza) [reply sw/applications/example_im2col/im2col_lib.c:165]: Good idea, w.r.t. the cited example I added PRINT_DEB for debugging purposes

File: sw/applications/example_im2col/im2col_lib.c [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+    for (int i=0; i<OH; i++)
+    {
+        for (int j=0; j<OW; j++)
+        {
+            #if DEBUG==2 || DEBUG==1
+            printf("%d ", output_data[i*OW + j]);
+            #endif
+        }
+        printf("\n");
+    }
```
[the remark this answers is not in the document]
Comment (TommiTerza) [reply sw/applications/example_im2col/im2col_lib.c]: Fixed!

File: sw/applications/example_im2col/im2col_lib.c [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+    // Return a 1 to indicate a success
+    return 1;
```
[the remark this answers is not in the document]
Comment (TommiTerza) [reply sw/applications/example_im2col/im2col_lib.c]: Yes, it's good practice, fixed

File: sw/applications/example_im2col/im2colGolden.c
[the remark this answers is not in the document]
Comment (TommiTerza) [reply sw/applications/example_im2col/im2colGolden.c:1]: I tried to do it in that way at first but the compiler gives a multiple declaration error that I wasn't able to fix

File: sw/applications/example_im2col/main.c [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    #if TIMING
+        CSR_READ(CSR_REG_MCYCLE, &cycles);
+    #endif
+    
+    errors = verify(NCHW_FORMAT);
+
+    PRINTF("im2col NCHW test executed\n");
+    
+    PRINTF_TIM("Total number of cycles: [%d]\n\n", cycles);
```
Comment (LuigiGiuffrida98) [inline sw/applications/example_im2col/main.c:42]: `.../sw/applications/example_im2col/main.c:51: undefined reference to 'PRINTF_TIM'`

Please, doublecheck the occurrences of `PRINTF_TIM`

File: sw/applications/example_im2col/im2col_lib.c [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    https://github.com/T-head-Semi/csi-nn2/blob/main/source/reference/im2col.c
+*/
+
+#include "im2col_lib.h"
+
+int output_data[OH_NCHW*OW_NCHW];
+
+int im2col_nchw_int32()
+{
+    PRINTF("%d %d\n", OH_NCHW, OW_NCHW);
```
Comment (LuigiGiuffrida98) [inline sw/applications/example_im2col/im2col_lib.c:22]: Prettyprint this as in line 87

**Review by LuigiGiuffrida98 — CHANGES_REQUESTED** [review]
I've executed the code on the pynq-z2 and it works even with the minimal configuration (2 BANKS), after these last updates it can be merged.

@davideschiavone please enable the CI checks on this PR so that we can check either there that everything works.

**Pull request merged**